catchup/universalFetcher_test.go

```diff
@@ -127,3 +131,136 @@ func TestUGetBlockUnsupported(t *testing.T) {
+	require.Equal(t, err.Error(), "processBlockBytes(22): cannot decode block from peer test: Unknown field: rn\x00 at Block")
+}
+
+// TestRequestBlockBytesErrors checks the error handling in requestBlockBytes
+func TestRequestBlockBytesErrors(t *testing.T) {
+
+	cfg := config.GetDefaultLocal()
+
+	ledger, next, _, err := buildTestLedger(t, bookkeeping.Block{})
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+
+	blockServiceConfig := config.GetDefaultLocal()
+	blockServiceConfig.EnableBlockService = true
+
+	net := &httpTestPeerSource{}
+
+	up := makeTestUnicastPeer(net, t)
+	ls := rpcs.MakeBlockService(logging.Base(), blockServiceConfig, ledger, net, "test genesisID")
+	ls.Start()
+
+	fetcher := makeUniversalBlockFetcher(logging.TestingLog(t), net, cfg)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	_, _, _, err = fetcher.fetchBlock(ctx, next, up)
+	require.Equal(t, err.Error(), "wsFetcherClient(test).requestBlock(1): Request failed, context canceled")
+
+	ctx = context.Background()
+
+	responseOverride := network.Response{Topics: network.Topics{network.MakeTopic(rpcs.BlockDataKey, make([]byte, 0))}}
+	up = makeTestUnicastPeerWithResponseOverride(net, t, &responseOverride)
+
+	_, _, _, err = fetcher.fetchBlock(ctx, next, up)
+	require.Equal(t, err.Error(), "wsFetcherClient(test): request failed: cert data not found")
+
+	responseOverride = network.Response{Topics: network.Topics{network.MakeTopic(rpcs.CertDataKey, make([]byte, 0))}}
+	up = makeTestUnicastPeerWithResponseOverride(net, t, &responseOverride)
+
+	_, _, _, err = fetcher.fetchBlock(ctx, next, up)
+	require.Equal(t, err.Error(), "wsFetcherClient(test): request failed: block data not found")
+
+}
+
+type TestHTTPHandler struct {
+	exceedLimit bool
+	status      int
+	content     []string
+}
+
+func (thh *TestHTTPHandler) ServeHTTP(response http.ResponseWriter, request *http.Request) {
+	for _, c := range thh.content {
+		response.Header().Add("Content-Type", c)
+	}
+	response.WriteHeader(thh.status)
+	bytes := make([]byte, 1)
+	if thh.exceedLimit {
+		bytes = make([]byte, 5242881)
+	}
+	response.Write(bytes)
+	return
+}
+
+// TestGetBlockBytesHTTPErrors tests the errors reported from getblockBytes for http peer
+func TestGetBlockBytesHTTPErrors(t *testing.T) {
+
+	cfg := config.GetDefaultLocal()
+	net := &httpTestPeerSource{}
+
+	ls := &TestHTTPHandler{}
+	//rpcs.MakeBlockService(logging.Base(), blockServiceConfig, ledger, net, "test genesisID")
+
+	nodeA := basicRPCNode{}
+	nodeA.RegisterHTTPHandler(rpcs.BlockServiceBlockPath, ls)
+	nodeA.start()
+	defer nodeA.stop()
+	rootURL := nodeA.rootURL()
+
+	net.addPeer(rootURL)
+	fetcher := makeUniversalBlockFetcher(logging.TestingLog(t), net, cfg)
+
+	ls.status = http.StatusBadRequest
+	_, _, _, err := fetcher.fetchBlock(context.Background(), 1, net.GetPeers()[0])
+	require.Regexp(t,
+		"getBlockBytes error response status code 400 when requesting .* Response body",
+		err.Error())
+
```

Comment:
While this is probably would work, a better way would be to create a custom error object and return it.
Then, you can verify that you received the expected error type rather then trying to compare it with the formatted text.